plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt

```diff
@@ -29,23 +28,30 @@ import com.ritense.plugin.exception.PluginDefinitionNotDeployedException
 import com.ritense.plugin.repository.PluginActionDefinitionRepository
 import com.ritense.plugin.repository.PluginActionPropertyDefinitionRepository
 import com.ritense.plugin.repository.PluginDefinitionRepository
+import com.ritense.plugin.repository.PluginPropertyRepository
 import mu.KotlinLogging
 import org.springframework.boot.context.event.ApplicationStartedEvent
-import org.springframework.transaction.event.TransactionalEventListener
+import org.springframework.context.event.EventListener
 import java.lang.reflect.Field
 import java.lang.reflect.Method
 import java.lang.reflect.Parameter
+import com.ritense.plugin.annotation.PluginProperty as PluginPropertyAnnotation
 
 class PluginDeploymentListener(
     private val pluginDefinitionResolver: PluginDefinitionResolver,
     private val pluginDefinitionRepository: PluginDefinitionRepository,
+    private val pluginPropertyRepository: PluginPropertyRepository,
     private val pluginActionDefinitionRepository: PluginActionDefinitionRepository,
     private val pluginActionPropertyDefinitionRepository: PluginActionPropertyDefinitionRepository
 ) {
 
-    @TransactionalEventListener(ApplicationStartedEvent::class)
+    @EventListener(ApplicationStartedEvent::class)
     fun deployPluginDefinitions() {
         logger.info { "Deploying plugins" }
+        pluginPropertyRepository.deleteAll()
```

Comment:
This causes a lot of unnecessary deletions, maybe change it so deletions only happen if there were changes?

Reply:
unfortunately, that won't fix the stackoverflow error. This error happens even when nothing changed at all: On every second application start-up the and the plugins are deployed for the second time, the error happens